src/AspNetCoreModuleV2/CommonLib/AppOfflineApplication.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#include "AppOfflineApplication.h"
+
+#include <experimental/filesystem>
+#include "HandleWrapper.h"
+
+HRESULT AppOfflineApplication::CreateHandler(IHttpContext* pHttpContext, IREQUEST_HANDLER** pRequestHandler)
+{
+    try
+    {
+        *pRequestHandler = new AppOfflineHandler(pHttpContext, m_strAppOfflineContent);
+    }
+    CATCH_RETURN();
+
+    return S_OK;
+}
+
+HRESULT AppOfflineApplication::OnAppOfflineFound()
+{
+    LARGE_INTEGER   li = {};
+
+    HandleWrapper<InvalidHandleTraits> handle = CreateFile(m_appOfflineLocation.c_str(),
+                        GENERIC_READ,
+                        FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,
+                        nullptr,
+                        OPEN_EXISTING,
+                        FILE_ATTRIBUTE_NORMAL,
+                        nullptr);
+
+    RETURN_LAST_ERROR_IF(handle == INVALID_HANDLE_VALUE);
+
+    RETURN_LAST_ERROR_IF (!GetFileSizeEx(handle, &li));
+
+    if (li.HighPart != 0)
```

Comment:
.NET Framework 4 limited the filesize to 1MB:

https://referencesource.microsoft.com/#System.Web/HttpRuntime.cs,264

I would have recommended matching the .NET Framework limit, however do we also need to now think about backwards compat with ANCM v1?  Since it allowed larger sizes, it would be a breaking change to reduce the size now.  I doubt many users would have an app_offline file larger than 1MB, but at the same time I don't know a good reason to limit the size to 1MB either.

Reply:
The .NET Framework code is a bit strange, in that it seems to treat an app_offline file larger than 1MB as if it doesn't exist.  I could understand ignoring the file content if over 1MB (or the portion over 1MB), but leaving the app online if app_offline is too large seems like a bug rather than intentional behavior, so we wouldn't want to match it in ANCM.